### PR TITLE
[logging] add more logging for failed payment shards

### PR DESCRIPTION
**Pull request by halseth**
Adding some logs in aid of debugging https://github.com/lightningnetwork/lnd/issues/4788

### CODE REVIEW

File: routing/payment_lifecycle.go [hunk 1]
```diff
@@ -14,6 +14,9 @@ import (
 	"github.com/lightningnetwork/lnd/routing/route"
 )
 
+// errShardHandlerExiting is returned fromt the shardHandler when it exits.
```
Comment (bhandras) [inline routing/payment_lifecycle.go:17]: s/fromt/from

File: routing/router.go [hunk 2]
```diff
@@ -2006,8 +2006,8 @@ func (r *ChannelRouter) tryApplyChannelUpdate(rt *route.Route,
 // processSendError analyzes the error for the payment attempt received from the
 // switch and updates mission control and/or channel policies. Depending on the
 // error type, this error is either the final outcome of the payment or we need
-// to continue with an alternative route. This is indicated by the boolean
-// return value.
+// to continue with an alternative route. A final outcome is indicated by a
```
Comment (carlaKC) [inline routing/router.go:2009]: unrelated to log bump?

**Review by bhandras — APPROVED** [review]
LGTM :+1:

**Review by carlaKC — APPROVED** [review]
LGTM!

File: routing/router.go (hunk 2 above)
Comment (halseth) [reply routing/router.go:2009]: Yeah, just noticed the doc hadn't been updated after the return value changed 😅

**Pull request merged**